### PR TITLE
Simple first pass on JSON-based metrics exporter

### PR DESCRIPTION
**Pull request by mason-stewart**
This PR was a result of asking the Encore team if there was a way to view metrics locally during development. The answer was "not yet", so I asked if they could provide some direction so that I could help get this implemented. @campesel gave some guidance, so here's a pull request as promised.

This also adds a development-only route at /__encore/metrics that renders the result of this exporter. This page will now render JSON response that look something like this:

```json
{
  "metrics": [
    {
      "name": "orders_processed",
      "type": "Counter",
      "labels": [],
      "value": 12
    },
    {
      "name": "e_requests_total",
      "type": "Counter",
      "labels": [
        {
          "endpoint": "Index"
        },
        {
          "code": "ok"
        }
      ],
      "value": 2
    }
  ]
}
```

Initially I wrote this just a few functions in the `runtime/appruntime/api/encore_routes.go` file, but it was clear that I was mimicking a lot of the functionality in the existing metrics exporters, and there was no simple, obvious (to me at least) place to test this if it was defined in `runtime/appruntime/api/encore_routes.go`, so I rewrote this to be a new exporter called `JSONBased`, following the existing patterns. 

I borrowed the tests from `runtime/appruntime/metrics/prometheus/prometheus_test.go` and adjusted them based on the shape of the JSON that @campesel requested. This was helpful since there were lots of little details I would have missed. I ended up borrowing heavily from the `getMetricData` in the `runtime/appruntime/metrics/prometheus/prometheus.go`.

This is enabled in development mode only (so far as I can understand) via a new field on `config.Config.Metrics`, but I believe it would be possible to enable this is non-local environments if desired.

I'm still a Golang rookie and I'm sure I've made some rookie mistakes here, so please don't hold back on feedback. Additionally, if the exporter-based approach is wrong, I'm happy to scrap this and take another approach. Cheers! 🤠

### CODE REVIEW

File: runtime/appruntime/metrics/json_based/json_based.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	"context"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"encore.dev/appruntime/config"
+	"encore.dev/metrics"
+)
+
+func New(svcs []string, cfg *config.JSONBasedMetricsProvider, rootLogger zerolog.Logger) *Exporter {
```
Comment (eandre) [inline runtime/appruntime/metrics/json_based/json_based.go:13]: I think the JSON metrics should not be thought of as a metrics provider/exporter, but as simply a pure function that takes `[]metrics.CollectedMetric` and returns a data a data structure. Essentially what you have with `(*Exporter).GetMetricData`. But we don't need to treat it as a "JSONBasedMetricsProvider" or change the configuration, I don't think. That way exposing metrics via API is orthogonal to the notion of a metrics exporter. (Thanks @campesel for pointing this out!)

File: runtime/appruntime/metrics/json_metrics_exporter.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package metrics
+
+import (
+	"encore.dev/appruntime/config"
+	"encore.dev/appruntime/metrics/json_based"
+)
+
+func init() {
+	registerProvider(providerDesc{
```
Comment (eandre) [inline runtime/appruntime/metrics/json_metrics_exporter.go:9]: See earlier comment: I think we shouldn't need the JSON metrics to be considered a metrics exporter

File: runtime/metrics/registry_internal.go [hunk 3]
```diff
@@ -68,6 +68,21 @@ func (r *Registry) Collect() []CollectedMetric {
...
 
+// Converts MetricType enum values into their common English names
+func (mt MetricType) Name() string {
+	switch mt {
+	case CounterType:
+		return "Counter"
+	case GaugeType:
+		return "Gauge"
+	case HistogramType:
+		return "Historgram"
```
Comment (eandre) [inline runtime/metrics/registry_internal.go:79]:
```suggestion
		return "Histogram"
```

**Review by eandre — COMMENTED** [review]
Thanks Mason for doing this! Overall it looks good, but a few points that would be nice to address before merging.

**Comment by mason-stewart** [comment]
Ah geez, I completely forgot about this PR for almost a year. Whoops. I'm going to close this since it has quite a few conflicts and we no longer need this. Thanks!

**Pull request closed without merging**